src/lib/agent/method/dapps/degen/unruggable/method/createMemecoin.ts

```diff
@@ -0,0 +1,41 @@
+import { CreateMemecoinParams } from 'src/lib/agent/schema';
+import { stark, uint256 } from 'starknet';
+import { DECIMALS, Entrypoint } from '../constant';
+import { rpcProvider } from 'src/lib/agent/starknetAgent';
+import { decimalsScale, execute } from '..';
+
+export const createMemecoin = async (
```

Comment:
here add validation for initial supply amount, current code just converts the string to BigInt without checks